src/recordlinker/linking/link.py

```diff
@@ -134,8 +140,11 @@ def compare(
     evals: list[models.BoundEvaluator] = algorithm_pass.bound_evaluators()
     # keyword arguments to pass to comparison functions
     kwargs: dict[typing.Any, typing.Any] = algorithm_pass.kwargs
-    # convert the Patient model into a PIIRecord for comparison
-    mpi_record: schemas.PIIRecord = schemas.PIIRecord.from_patient(patient)
+    # convert the Patient model into a cleaned PIIRecord for comparison
+    # FIXME: can we clean higher up on the stack?
+    mpi_record: schemas.PIIRecord = clean.clean(
+        schemas.PIIRecord.from_patient(patient), skip_values
+    )
```

Comment:
I think we could move the cleaning to `link_record_against_mpi` (line 269) where we are iterating through `pat in pats`. We then wouldn't have to pass skip values to `compare` and instead just pass `cleaned_record` and `cleaned_mpi_record` which would now also be a PIIRecord instead of `models.Patients`

Reply:
I want to do something like that, but the problem right now is we're still using the MPI patient model in this compare method (see line 151).  However, we are using it just for logging, so if we remove the extra logging detail we could rewrite the compare method to accept two PIIRecords rather than a PIIRecord (incoming) and Patient (existing) object.  What do you think, remove the logging to speed things up a bit?

Reply:
I want to make sure I understand the purpose of logging `Patient.reference_id`. We're doing that so that we can track all of the Patients an incoming record was compared to, right, not just the final one it was linked to?

Reply:
Yeah, it's just debugging info to help with a potential investigation if we ever wanted to dig into the details of how/why results were being returned.

Reply:
In that case, I think it is okay to remove so that both `record` and `patient` are PIIRecords.

Reply:
cool, I'll give this a try

Reply:
So I just made the change, https://github.com/CDCgov/RecordLinker/pull/291/commits/af509f0d284445c64209eaf54db87b4084e116ad, but I'm rethinking its value.

With multiple passes, its possible that there will be some overlap in the MPI records returned.  In fact, we highlighted this in some of the examples we put together.  However, the way its implemented now, we have to "re-clean" the MPI Patient data everytime its pulled back from the database (so every pass).  In other words, I think we end up calling the clean method the same number of times, just in a different place.

I think there is an upside to the new approach though, as now all cleaning is done in one method, `link` and it's not split between the two.

Reply:
Gotcha. I think the consolidation to one method and passing in the same data type (PIIRecord) seem like improvements to me.